torchtune/modules/transformer.py

```diff
@@ -125,19 +251,28 @@ class TransformerDecoder(nn.Module):
 
     def __init__(
         self,
+        *,
         tok_embeddings: nn.Embedding,
-        layer: TransformerDecoderLayer,
-        num_layers: int,
+        layer: Union[nn.Module, List[nn.Module]],
```

Comment:
do we need this flexibility, or can we simplify it by always creating the list internally?